app/src/main/java/com/example/nvdovin/weatherapp/domain/factory/GreenDaoFactory.java

```diff
@@ -0,0 +1,158 @@
+                .unique();
+    }
+
+    public List<WeatherData> getWeatherDataListByDTs(Long[] timestampArray, Long cityId) {
+
+        List<WeatherData> weatherDataList = new ArrayList<>();
+        for (int i = 0; i < timestampArray.length; i++) {
+            WeatherData weatherData = getWeatherDataByDT(timestampArray[i], cityId);
+            if (weatherData != null) {
+                weatherDataList.add(weatherData);
+            }
+        }
+        return weatherDataList;
+    }
+
+    public List<WeatherData> getWeatherDataForDay(Long cityId, Long dt, Long period) {
+        return daoSession.getWeatherDataDao().queryBuilder()
+                .where(WeatherDataDao.Properties.CityId.eq(cityId), WeatherDataDao.Properties.Dt.ge(dt),
+                        WeatherDataDao.Properties.Dt.le(dt + period))
+                .list();
+
+    }
+
+    public List<CityForecast> loadCityWeatherForNow(List<City> cities) {
+        Long currentTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+
+        List<CityForecast> cityForecastList = new ArrayList<>();
+        for (City city : cities) {
+            CityForecast cityForecast = new CityForecast();
+            cityForecast.setCityId(city.getId());
+            cityForecast.setCityName(city.getName());
+            WeatherData weatherDataForNow = daoSession.getWeatherDataDao().queryBuilder().where(
+                    new WhereCondition.StringCondition(context.getString(R.string.time_query),
+                            String.valueOf(cityForecast.getCityId()), String.valueOf(currentTime)
+                    )).unique();
+            cityForecast.setCurrentCityWeather(weatherDataForNow);
+            cityForecastList.add(cityForecast);
+        }
+        return cityForecastList;
+    }
+
+    public WeatherData getUnique(Long cityId, long time) {
+        return daoSession
+                .getWeatherDataDao()
+                .queryBuilder()
+                .where(
+                        new WhereCondition.StringCondition(context.getString(R.string.time_query),
+                                String.valueOf(cityId), String.valueOf(time)
+                        )).orderDesc(WeatherDataDao.Properties.Dt).limit(1).unique();
+    }
+
+    public List<DailyCityForecast> getDailyCityForecast(Long cityId, Long dt, int numberOfDays) {
+        List<DailyCityForecast> dailyCityForecastList = new ArrayList<>();
+        DailyCityForecast dailyCityForecast;
+        for (int a = 0; a < numberOfDays; a++) {
+
+            int tempMax = 178;
+            int tempMin = 378;
+            dailyCityForecast = new DailyCityForecast(getUnique(cityId, dt));
+            dailyCityForecast.setCityId(cityId);
+            dailyCityForecast.setDay(new Date(dt * 1000));
+
+            City city = getCityById(cityId);
+            dailyCityForecast.setCityName(city.getName());
+            List<WeatherData> weatherList = getWeatherDataForDay(cityId, dt, 86400L);
```

Comment:
You have declared on top DAY_IN_MILLISECONDS_INCLUSIVE, why not to use it here?

And do the same please with others hardcoded values from this method.(178, 378, 1000)